src/ppx/instrument.ml

```diff
@@ -490,11 +490,14 @@ struct
         |> List.map (fun (location_trace, p'') ->
           (location_trace, Pat.alias ~loc ~attrs p'' x))
 
-      | Ppat_construct (c, Some p') ->
+      | Ppat_construct (c, Some ([], p')) ->
         recur ~enclosing_loc p'
         |> List.map (fun (location_trace, p'') ->
           (location_trace, Pat.construct ~loc ~attrs c (Some p'')))
 
+      | Ppat_construct (_, Some (_ :: _, _)) ->
+        Location.raise_errorf ~loc "bisect_ppx: named existentials aren't supported"
```

Comment:
Thanks. Is there any reason not to support these? Could you link to a reference on them?

Reply:
Hi @aantron, thanks for reviewing!

> Is there any reason not to support these?

No, there's no reason at all not to support them. I should have explained the following in the PR message: I myself don't know all >100 ppxlib reverse dependencies well enough to decide myself what the behavior should be when encountering the new feature in the AST.

If you let me know what the correct behavior would look like, I can update the PR. As a guess: maybe in `bound_variables` and in `has_polymorphic_variant`, the naming of the existentials should simply be ignored; and in `rotate_or_patterns_to_top`, the existential in the return pattern should be named in the same way as in the input pattern? That guess might be wrong though. What would you say as the maintainer?

> Could you link to a reference on them?

Sure. There's
- a section in the OCaml manual: https://ocaml.org/manual/gadts-tutorial.html#s:explicit-existential-name
- a short syntax summary on the PR: https://github.com/ocaml/ocaml/pull/9584
- a longer explanation on an issue: https://github.com/ocaml/ocaml/issues/7074